src/wallet/wallet.ts

```diff
@@ -443,28 +580,32 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
    * @memberof HathorWalletServiceWallet
    * @inner
    */
-  signMessage(timestamp: number): string {
-    const xpriv = walletUtils.getXPrivKeyFromSeed(this.seed, {passphrase: this.passphrase, networkName: this.network.name});
-    const derivedPrivKey = walletUtils.deriveXpriv(xpriv, '0\'');
-    const address = derivedPrivKey.publicKey.toAddress(this.network.getNetwork()).toString();
-
-    const message = new bitcore.Message(String(timestamp).concat(this.walletId!).concat(address));
+  signMessage(xpriv: bitcore.HDPrivateKey, timestamp: number, walletId: string): string {
+    const address = xpriv.publicKey.toAddress(this.network.getNetwork()).toString();
+    const message = new bitcore.Message(String(timestamp).concat(walletId).concat(address));
 
-    return message.sign(derivedPrivKey.privateKey);
+    return message.sign(xpriv.privateKey);
   }
 
   /**
    * Validate that the wallet auth token is valid
    * If it's not valid, requests a new one and update
    *
+   * @param {string} usePassword Accepts the password as a parameter so we don't have to ask
+   * the client for it if we already have it in memory
+   *
    * @memberof HathorWalletServiceWallet
    * @inner
    */
-  async validateAndRenewAuthToken() {
+  async validateAndRenewAuthToken(usePassword?: string) {
+    if (!this.walletId) {
+      throw new Error('Wallet not ready yet.');
+    }
+
     const now = new Date();
     const timestampNow = Math.floor(now.getTime() / 1000);
 
-    const validateJWTExpireDate = (token) => {
+    const validateJWTExpireDate = (token: string) => {
```

Comment:
Missing return type?

Reply:
Typescript infers the type from the context there, but we can be explicit about it!

Done in d666b6e8a8f19db7c2650388fa31d8f24c1655ca

Reply:
Thanks! The downside of type inference is that the linter also infers `const x = myfunction()`, and it might propagate over and over. I thought we had a linter in this repository. If not, would you please create an issue to get it done?

Reply:
Issue created at https://github.com/HathorNetwork/hathor-wallet-lib/issues/321